apps/src/templates/instructions/ContainedLevelResetButton.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from '@cdo/apps/templates/Button';
+import HelpTip from '@cdo/apps/lib/ui/HelpTip';
+import {CourseRoles} from '@cdo/apps/templates/currentUserRedux';
+import {resetContainedLevel} from '@cdo/apps/code-studio/levels/codeStudioLevels';
+import {connect} from 'react-redux';
+import {queryUserProgress} from '@cdo/apps/code-studio/progressRedux';
+import i18n from '@cdo/locale';
+import experiments from '@cdo/apps/util/experiments';
+
+export const UnconnectedContainedLevelResetButton = ({
+  userId,
+  queryUserProgress,
+  hasLevelResults,
+  userRoleInCourse,
+  codeIsRunning
+}) => {
+  const isEnabled = experiments.isEnabled('instructorPredictLevelReset');
```

Comment:
we've run into performance problems in the past in situations where `experiments.isEnabled` is called frequently, in particular it caused big slowdowns for some users when we added it to the rendering of progress bubbles. Maybe just check that this doesn't substantially slow down resizing of the instructions tab?

Reply:
oh good call -- I'll check!

Reply:
it is not checked on instruction resize. It is re-checked when run is clicked -- I ran and reset a few times very quickly and there was a slight delay in logging but nothing really noticeable in the UI. This seems like a good thing to move to `useMemo` if there are any concerns though so I'll do that